### PR TITLE
Implement fn:matches

### PR DESCRIPTION
**Pull request by DrRataplan**
This works by requiring xspattern in two different ways: in unit tsts from node_modules, and in a
build using an UMD, and exposing it on window. Two different tsconfigs make this complete.

### CODE REVIEW

**Comment by DrRataplan** [comment]
Not to be integrated yet. There is a bug: `fn:matches('aba', 'b')` should return true, while it is not

File: src/expressions/functions/builtInFunctions.string.ts [hunk 1]
```diff
@@ -792,6 +817,14 @@ export default {
 			localName: 'codepoint-equal',
 			namespaceURI: FUNCTIONS_NAMESPACE_URI,
 			returnType: 'xs:boolean?'
+		},
+
+		{
+			argumentTypes: ['xs:string?', 'xs:string'],
```
Comment (JosVerburg) [inline src/expressions/functions/builtInFunctions.string.ts]: Could we also add the 2nd variant (with flags)?

File: src/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import IDocumentWriter from './documentWriter/IDocumentWriter';
+
```
Comment (JosVerburg) [inline src/index.ts]: Accidental enter hit?

File: src/expressions/functions/builtInFunctions.string.ts (hunk 1 above)
Comment (DrRataplan) [reply src/expressions/functions/builtInFunctions.string.ts]: not easily no, I'd like to delay the flags for now and focus on a first push for the 2-arity matches

**Pull request merged**